Split into smaller packages

jeromelaban opened the pull request:
## PR Type
What kind of change does this PR introduce?
- Refactoring (no functional changes, no api changes)

## What is the new behavior?

This change introduces:
- Uno.Core.Extensions
- Uno.Core.Extensions.Collections
- Uno.Core.Extensions.Disposables
- Uno.Core.Extensions.Equality
- Uno.Core.Extensions.Logging
- Uno.Core.Extensions.Logging.Singleton
- Uno.Core.Extensions.Threading

Splitting the code from the Uno.Core assembly into separate, smaller packages with less dependencies.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno.Core/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno.Core/blob/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. 
     Please note that breaking changes are likely to be rejected -->


## Other information
<!-- Please provide any additional information if necessary -->

Internal Issue (If applicable):
<!-- Link to relevant internal issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

carldebilly reviewed (approved):
TL;JA

dr1rrb commented on src/Uno.Core/Collections/PairExtensions.cs:
```diff
@@ -0,0 +1,124 @@
+	public static class ListExtensions
+	{
+
+		/// <summary>
+		/// Adapts a list of type T into a list of type U
+		/// </summary>
+		/// <typeparam name="T">The type to adapt.</typeparam>
+		/// <typeparam name="U">The target type.</typeparam>
+		/// <param name="items">The list to adapt</param>
+		/// <param name="from">The function used to adapt a U into a T.</param>
+		/// <param name="to">The function used to adapt a T into a U.</param>
+		/// <returns>A adapted list of the target type.</returns>
+		public static IList<U> Adapt<T, U>(this IList<T> items, Func<U, T> from, Func<T, U> to)
+		{
+			return new ListAdapter<T, U>(items, from, to);
+		}
+
+		public static IList<U> Adapt<T, U>(this IList<T> items)
+		{
+			return Adapt(items, Funcs<U, T>.Convert, Funcs<T, U>.Convert);
+		}
+	}
+
+	public static class CollectionsExtensions
+	{
+		public static ICollection<U> Adapt<T, U>(this ICollection<T> collection)
+		{
+			return Adapt(collection, Funcs<U, T>.Convert, Funcs<T, U>.Convert);
+		}
+
+		/// <summary>
+		/// Adapts a collection of type T into a collection of type U
+		/// </summary>
+		/// <typeparam name="TSource">The type to adapt.</typeparam>
+		/// <typeparam name="TTarget">The target type.</typeparam>
+		/// <param name="collection">The collection to adapt</param>
+		/// <param name="from">The function used to adapt a U into a T.</param>
+		/// <param name="to">The function used to adapt a T into a U.</param>
+		/// <returns>A adapted collection of the target type.</returns>
+		public static ICollection<TTarget> Adapt<TSource, TTarget>(this ICollection<TSource> collection, Func<TTarget, TSource> from, Func<TSource, TTarget> to)
```
```suggestion
		/// <summary>
		/// Adapts a list of type <typeparamref name="TSource"/> into a list of type <typeparamref name="TTarget"/>.
		/// </summary>
		/// <typeparam name="TSource">The type to adapt.</typeparam>
		/// <typeparam name="TTarget">The target type.</typeparam>
		/// <param name="items">The list to adapt</param>
		/// <param name="from">The function used to adapt a TTarget into a TSource.</param>
		/// <param name="to">The function used to adapt a TSource into a TTarget.</param>
		/// <returns>A adapted list of the target type.</returns>
		public static IList<TTarget> Adapt<TSource, TTarget>(this IList<TSource> items, Func<TTarget, TSource> from, Func<TSource, TTarget> to)
		{
			return new ListAdapter<TSource, TTarget>(items, from, to);
		}

		public static IList<TTarget> Adapt<TSource, TTarget>(this IList<TSource> items)
		{
			return Adapt(items, Funcs<TTarget, TSource>.Convert, Funcs<TSource, TTarget>.Convert);
		}
	}

	public static class CollectionsExtensions
	{
		public static ICollection<TTarget> Adapt<TSource, TTarget>(this ICollection<TSource> collection)
		{
			return Adapt(collection, Funcs<TTarget, TSource>.Convert, Funcs<TSource, TTarget>.Convert);
		}

		/// <summary>
		/// Adapts a collection of type <typeparamref name="TSource"/> into a collection of type <typeparamref name="TTarget"/>.
		/// </summary>
		/// <typeparam name="TSource">The type to adapt.</typeparam>
		/// <typeparam name="TTarget">The target type.</typeparam>
		/// <param name="collection">The collection to adapt</param>
		/// <param name="from">The function used to adapt a TTarget into a TSource.</param>
		/// <param name="to">The function used to adapt a TSource into a TTarget.</param>
		/// <returns>A adapted collection of the target type.</returns>
		public static ICollection<TTarget> Adapt<TSource, TTarget>(this ICollection<TSource> collection, Func<TTarget, TSource> from, Func<TSource, TTarget> to)
```

jeromelaban replied:
This is legacy code, no particular need to touch this at this point :)

jeromelaban committed:
feat: Split Uno.Core into smaller packages with less dependencies

New packages are :
- Uno.Core.Extensions
- Uno.Core.Extensions.Compatibility
- Uno.Core.Extensions.Collections
- Uno.Core.Extensions.Disposables
- Uno.Core.Extensions.Equality
- Uno.Core.Extensions.Logging
- Uno.Core.Extensions.Logging.Singleton
- Uno.Core.Extensions.Threading

jeromelaban committed:
docs: Update readme after packages split
